### DISCUSSION TITLE
Create custom field in dataset creation form.

### DISCUSSION THREAD

**Discussion by mitthuzyxware** [GetDKAN/dkan#4077] [General]
Hi,

I am reaching out regarding my work on the DKAN 2.0 module within Drupal 10 and the frontend is decoupled using React, specifically in our requirement to introduce a new field into the dataset form.

As part of this process, I successfully added a custom field to the DATA content type. However, we encountered an issue where the custom field values are not being retrieved in the schema APIs. Upon inspecting the module, I observed that fields are typically added via schema from JSON-formatted files.

I'm reaching out to inquire if there is an alternative method to create a custom field so that its values can be obtained in the same metastore APIs we are currently utilizing by default.

Any guidance or suggestions you could provide would be greatly appreciated. Thank you for your time and assistance.

**Comment by janette** [comment]
@mitthuzyxware to add a custom field to the form you will need to copy the entire schema/collections directory out of dkan to your docroot directory (docroot/schema/collections). Then you can edit the dataset.json schema to add your field under "properties".

```
"myNewField": {
  "title": "Custom Field",
  "description": "Some descriptive text.",
  "type": "string"
},
```

**Reply by mitthuzyxware** [reply]
@janette 
Many thanks for your response.
I found about the field creation but it was good to know that we can move the schema/collection files out of dkan so now I can avoid any extra patch creation.
